### PR TITLE
Forch varz interface with basic varz

### PR DESCRIPTION
**Pull request by anurag6**

### CODE REVIEW

**Comment by grafnu** [comment]
Looks good overall -- should also get a minimal unit test in here, just enough to validate that it's working.

File: debian/control [hunk 1]
```diff
@@ -26,6 +26,7 @@ Architecture: all
 Depends: python3-faucet (= FAUCET_VERSION),
          python3-netaddr,
          python3-networkx,
+         python3-prometheus-client (>= 0.7.1), python3-prometheus-client (<< 0.7.2),
```
Comment (grafnu) [inline debian/control]: I think you can just do = for exact match

File: forch/forch_metrics.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    def stop(self):
+        """Kill varz server"""
+        self._http_server.stop_server()
+
+    def update_var(self, var, value):
+        """Update given varz with new value"""
+        varz = self._metrics.get(var)
+        if not varz:
+            LOGGER.error('Error updating to varz %s since it is not known.', var)
```
Comment (grafnu) [inline forch/forch_metrics.py]: seems like this should be a hard error (throw exception)

File: forch/forch_metrics.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        if isinstance(varz, Info):
+            varz.info(value)
+        else:
+            LOGGER.error('Error updating to varz %s since it\'s type %s is not known.',
+                         var, type(varz))
+            return
+
+    def _add_var(self, var, var_help, metric_type):
+        """Add varz to be tracked"""
+        self._metrics[var] = metric_type(var, var_help, registry=self._reg)
```
Comment (grafnu) [inline forch/forch_metrics.py]: again hard error

File: forch/forchestrator.py [hunk 4]
```diff
@@ -99,6 +100,8 @@ def initialize(self):
 
         self._validate_config_files()
 
+        self._forch_metrics = ForchMetrics(8302, self._config.http)
```
Comment (grafnu) [inline forch/forchestrator.py]: this is not right -- there should only be one config block for this, and the port should be part of it.  I think we might need to do an audit of the config setup -- The current http config makes no sense for this (it's specifying a file root).  If you want to simplify and not make things worse, find a way to do this without passing self._config.http in.

Comment (anurag6) [reply forch/forchestrator.py]: Makes sense. Let me look into it.

File: .travis.yml [hunk 5]
```diff
@@ -9,13 +9,11 @@ addons:
   apt:
     update: true
```
Comment (anurag6) [inline .travis.yml]: Will remove once travis tests go through

File: forch/forch_metrics.py (hunk 2 above)
Comment (anurag6) [reply forch/forch_metrics.py]: Done

File: forch/forch_metrics.py (hunk 3 above)
Comment (anurag6) [reply forch/forch_metrics.py]: Done.

File: forch/forchestrator.py (hunk 4 above)
Comment (anurag6) [reply forch/forchestrator.py]: Changed now.

File: debian/control (hunk 1 above)
Comment (anurag6) [reply debian/control]: Done

**Comment by anurag6** [comment]
@grafnu  PTAL.

File: debian/control [hunk 6]
```diff
@@ -26,6 +26,7 @@ Architecture: all
 Depends: python3-faucet (= FAUCET_VERSION),
          python3-netaddr,
          python3-networkx,
+         python3-prometheus-client (= 0.7.1), python3-prometheus-client (<< 0.7.2),
```
Comment (grafnu) [inline debian/control]: why << 0.7.2?

File: forch/forchestrator.py [hunk 7]
```diff
@@ -99,6 +100,8 @@ def initialize(self):
 
         self._validate_config_files()
 
+        self._forch_metrics = ForchMetrics(8302)
```
Comment (grafnu) [inline forch/forchestrator.py]: port should be configurable

File: debian/control (hunk 6 above)
Comment (anurag6) [reply debian/control]: Changed

File: forch/forchestrator.py (hunk 7 above)
Comment (anurag6) [reply forch/forchestrator.py]: Done.

File: debian/control [hunk 8]
```diff
@@ -26,7 +26,7 @@ Architecture: all
 Depends: python3-faucet (= FAUCET_VERSION),
          python3-netaddr,
          python3-networkx,
-         python3-prometheus-client (= 0.7.1), python3-prometheus-client (<< 0.7.2),
+         python3-prometheus-client (= 0.7.1),,
```
Comment (grafnu) [inline debian/control]: extra comma

File: forch/http_server.py [hunk 9]
```diff
@@ -78,7 +78,10 @@ def __init__(self, port, config=None):
         self._config = config
         self._paths = {}
         self._server = None
-        self._root_path = config.http_root if config else 'public'
+        if config and config.http_root:
```
Comment (grafnu) [inline forch/http_server.py]: self._root_path = config.http_root if config and config.http_root else 'public'

File: proto/forch_configuration.proto [hunk 10]
```diff
@@ -150,3 +153,11 @@ message EventClientConfig {
   // port debounce time in seconds
   int32 port_debounce_sec = 1;
 }
+
+/*
+ * Varz interface section in Forch configuration
+ */
+message VarzInterfaceConfig {
+  // port debounce time in seconds
```
Comment (grafnu) [inline proto/forch_configuration.proto]: ?

File: forch/forch_metrics.py [hunk 11]
```diff
@@ -8,14 +8,16 @@
...
 
 class ForchMetrics():
     """Class that implements the module that exposes varz for metrics"""
     _reg = REGISTRY
 
-    def __init__(self, local_port):
-        self._local_port = local_port
+    def __init__(self, varz_config):
+        self._local_port = varz_config.varz_port or DEFAULT_VARZ_PORT
+        LOGGER.info('Anurag forch_metrics port is %s', self._local_port)
```
Comment (grafnu) [inline forch/forch_metrics.py]: reminder to remove Anurag (I'd still leave the logging statement)

Comment (anurag6) [reply forch/forch_metrics.py]: Yeah my bad.

File: debian/control (hunk 8 above)
Comment (anurag6) [reply debian/control]: Done

File: forch/http_server.py (hunk 9 above)
Comment (anurag6) [reply forch/http_server.py]: Done.

File: proto/forch_configuration.proto (hunk 10 above)
Comment (anurag6) [reply proto/forch_configuration.proto]: Done.

File: forch/forch_metrics.py (hunk 11 above)
Comment (anurag6) [reply forch/forch_metrics.py]: Done.

**Pull request merged**